code.py

```diff
@@ -69,20 +69,25 @@ def nextDist(self, pose, line, index_bot):
 		
 		if d1 < d2:
 			self.pose[index_bot] = [x2, y2]
+			return d1
 		else:
 			self.pose[index_bot] = [x1, y1]
-		dist = min(d1,d2)
-		return dist
+			return d2
 
 
 	# Heuristics func in paper: Return distance/cost of a single input group
-	def groupDist(self, sol):
+	def groupDist(self, group):
 		index_line = 0
-		cost = 0
-		for index_bot in sol:
+		c = np.empty(len(self.pose))
+		for i in range(len(c)):
+			c[i] = 0
+			
+		for index_bot in group:
 			dist = self.nextDist(pose = self.pose[index_bot], line = self.lines[index_line], index_bot = index_bot)
-			cost = cost + dist + self.length[index_line]
-			index_line = index_line + 1		
+			c[index_bot] = c[index_bot] + dist + self.length[index_line]
+			index_line = index_line + 1
+			
+		cost = max(c)
```

Comment:
One group will have only one cost. max() function is hence redundant here.

Reply:
Each group will have 3 costs for each bot. Cost is just returning the max out of those 3 values

Reply:
Each individual should have 3 groups, each group should have one cost.

Reply:
Yeah. I have read and understood the paper. By tomorrow, I will upload a draft of the code as given in the paper